Geometry_Engine/Query/GeometryHash.cs

```diff
@@ -0,0 +1,427 @@
+
+            if (!skipLastPoint)
+                hash = hash.Concat(curve.EndPoint().ToDoubleArray(translationFactor));
+
+            return hash.ToArray();
+        }
+
+        /***************************************************/
+
+        [Description("The GeometryHash for an Circle is calculated as the GeometryHash of the start, 1/3rd and 2/3rd points of the Circle.")]
+        private static double[] GeometryHash(this Circle curve, double translationFactor, bool skipLastPoint = false)
+        {
+            translationFactor += (int)TypeTranslationFactor.Circle;
+
+            IEnumerable<double> hash = curve.StartPoint().ToDoubleArray(translationFactor)
+                   .Concat(curve.PointAtParameter(0.33).ToDoubleArray(translationFactor));
+
+            if (!skipLastPoint)
+                hash = hash.Concat(curve.PointAtParameter(0.66).ToDoubleArray(translationFactor));
+
+            return hash.ToArray();
+        }
+
+        /***************************************************/
+
+        [Description("The GeometryHash for an Ellipse is calculated as the GeometryHash of the start, 1/3rd and 2/3rd points of the Ellipse.")]
+        private static double[] GeometryHash(this Ellipse curve, double translationFactor, bool skipLastPoint = false)
+        {
+            translationFactor += (int)TypeTranslationFactor.Ellipse;
+
+            IEnumerable<double> hash = curve.StartPoint().ToDoubleArray(translationFactor)
+               .Concat(curve.PointAtParameter(0.33).ToDoubleArray(translationFactor));
+
+            if (!skipLastPoint)
+                hash = hash.Concat(curve.PointAtParameter(0.66).ToDoubleArray(translationFactor));
+
+            return hash.ToArray();
+        }
+
+        /***************************************************/
+
+        [Description("The GeometryHash for a Line is calculated as the GeometryHash of the start and end point of the Line.")]
+        private static double[] GeometryHash(this Line curve, double translationFactor, bool skipLastPoint = false)
+        {
+            translationFactor += (int)TypeTranslationFactor.Line;
+
+            if (skipLastPoint)
+                return curve.StartPoint().ToDoubleArray(translationFactor);
+
+            return curve.StartPoint().ToDoubleArray(translationFactor)
+               .Concat(curve.EndPoint().ToDoubleArray(translationFactor))
+               .ToArray();
+        }
+
+        /***************************************************/
+
+        [Description("The GeometryHash for a NurbsCurve is obtained by getting moving the control points " +
+            "by a translation factor composed by the weights and a subarray of the knot vector. " +
+            "The subarray is made by picking as many elements from the knot vector as the curve degree value.")]
+        private static double[] GeometryHash(this NurbsCurve curve, double translationFactor, bool skipLastPoint = false)
```

Comment:
Just noting here that we are not doing anything with the argument for the general case here, which I think probably is fine, as it can be a bit more complicated to do. Just putting as a review comment for myself that I have looked at it :)